modules/@apostrophecms/modal/ui/apos/mixins/AposDocsManagerMixin.js

```diff
@@ -122,20 +153,19 @@ export default {
     // a computed prop rather than a method call in the template.
     maxReached() {
       // Reaching max and exceeding it are different things
-      const result = this.relationshipField.max && this.checked.length >= this.relationshipField.max;
-      return result;
+      return this.relationshipField.max &&
+        this.checked.length >= this.relationshipField.max;
     },
     selectAll() {
       if (!this.checked.length) {
-        this.items.forEach((item) => {
-          const relationshipsMaxedOrUnpublished = this.relationshipField &&
-          (this.maxReached() || !item.lastPublishedAt);
 
-          if (relationshipsMaxedOrUnpublished) {
+        this.items.forEach((item) => {
+          const notPublished = this.manuallyPublished && !item.lastPublishedAt;
+          if (this.relationshipField && (this.maxReached() || notPublished)) {
             return;
           }
 
-          this.checked.push(item._id);
+          this.addCheckedDoc(item);
         });
```

Comment:
The main fix is in this `selectAll` method. Before manually published doc were impossible to check. 
And `checkedDocs` as no updated because the `checked.push` wasn't triggering the watcher..